src/metabase/embed/settings.clj

```diff
@@ -32,39 +37,67 @@
   :visibility :authenticated
   :export?    true
   :audit      :getter
-  :setter     (fn [new-value]
-                (when (not= new-value (setting/get-value-of-type :boolean :enable-embedding))
-                  (setting/set-value-of-type! :boolean :enable-embedding new-value)
-                  (when (and new-value (str/blank? (embed/embedding-secret-key)))
-                    (embed/embedding-secret-key! (crypto-random/hex 32)))
-                  (snowplow/track-event! ::snowplow/embed_share
-                                         {:event                      (if new-value
-                                                                        :embedding-enabled
-                                                                        :embedding-disabled)
-                                          :embedding-app-origin-set   (boolean (embedding-app-origin))
-                                          :number-embedded-questions  (t2/count :model/Card :enable_embedding true)
-                                          :number-embedded-dashboards (t2/count :model/Dashboard :enable_embedding true)}))))
+  :deprecated "0.51.0"
+  :setter     (make-embedding-toggle-setter :enable-embedding "embedding"))
 
-(defsetting enable-embedding-static
-  (deferred-tru "Allow admins to embed Metabase via static embedding?")
+(defsetting embedding-app-origin
+  (deferred-tru "Allow this origin to embed the full Metabase application.")
+  ;; This value is usually gated by [[enable-embedding]]
+  :feature    :embedding
+  :type       :string
+  :export?    false
+  :visibility :public
+  :audit      :getter
+  :encryption :no)
```

Comment:
@escherize I'm not sure this is the correct setting for `encryption` - I saw that it was `:never` but I was running into errors

Reply:
Looks like that defsetting value got changed on master, so you're right that `:no` is correct. Thank you!